clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -2074,6 +2083,35 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
       return true;
     break;
   }
+  case Builtin::BI__builtin_hlsl_splitdouble: {
+    if (SemaRef.checkArgCount(TheCall, 3))
+      return true;
+
+    Expr *Op0 = TheCall->getArg(0);
+
+    auto CheckIsNotDouble = [](clang::QualType PassedType) -> bool {
```

Comment:
Naming convention for `CheckIsNotDouble` seems reversed from what I'd expect based on some of the established patterns, since double is the desired type.  See similar functions `CheckAllArgsHaveFloatRepresentation`, `CheckFloatOrHalfRepresentations`, `CheckNoDoubleVectors`, `CheckFloatingOrIntRepresentation`, etc...

Note: same comment for function `CheckIsNotUint` below.